package/kedro_viz/server.py

```diff
@@ -452,10 +452,25 @@ def format_pipeline_data(
                 edges_list.append(edge)
             namespace_tags[namespace].update(node.tags)
             node_dependencies[task_id].add(namespace_id)
-
+    # Parameters and data
     for namespace, tag_names in sorted(namespace_tags.items()):
         is_param = bool("param" in namespace.lower())
         node_id = _hash(namespace)
+        if not is_param:
+
+            if KEDRO_VERSION.match(">=0.16.0"):
+                try:
+                    # pylint: disable=protected-access
+                    dataset = _CATALOG._get_dataset(namespace)
+                    _NODES[node_id] = {"type": "data", "obj": dataset}
```

Comment:
This line and line 472 are complete duplicates. I think it can fairly easily be refactored to avoid the repetition.